src/hooks/useMutation.ts

```diff
@@ -0,0 +1,66 @@
+import { queryApi } from "../client";
+import { DocumentNode, GraphQLFormattedError } from "graphql";
+import { useState, useEffect, useCallback, useRef } from "react";
+import { useExtension } from "./ExtensionProvider";
+
+type MutationError = GraphQLFormattedError | Error;
+type Response<T> = { data: T | null; errors?: GraphQLFormattedError[] } | null;
+
+export const useMutation = <T, V>(
+  query: DocumentNode
+): [
+  (variables: V) => Promise<Response<T>>,
+  {
+    data: T | null;
+    loading: Boolean;
+    error?: MutationError;
+  }
+] => {
+  const { id, port } = useExtension();
+  const mounted = useRef(true);
+  const [data, setData] = useState<T | null>(null);
```

Comment:
What is the `T | null` part?

Reply:
`T` is a generic, meaning this shape is passed into the method as a type. I am going to rename these, a common convention is making these one-letter variables, but kinda leaning more this way now.

https://twitter.com/kentcdodds/status/1348016701632221184?s=20